### PR TITLE
build: adopt black codestyle

### PR DESCRIPTION
**Pull request by FFY00**
On top of #121

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -12,6 +12,12 @@ repos:
   - id: debug-statements
   - id: end-of-file-fixer
   - id: trailing-whitespace
+  - id: double-quote-string-fixer
+- repo: https://github.com/psf/black
+  rev: 20.8b1
+  hooks:
+  - id: black
+    args: ["-S", "-l", "127"]
```
Comment (gaborbernat) [inline .pre-commit-config.yaml:20]: Please add these arguments into the pyproject.toml config, so tools can pick it up autoamtically. Any reason you went with 127 instead of 120?

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (FFY00) [reply .pre-commit-config.yaml]: 127 is the size of the Github editor.

Comment (gaborbernat) [reply .pre-commit-config.yaml]: fair

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,8 @@
 [build-system]
 requires = ['setuptools >= 40.8.0', 'wheel']
 build-backend = 'setuptools.build_meta'
+
+[tool.black]
+line-length = 127
+skip-string-normalization = true
+target-version = ['py27', 'py35', 'py36', 'py37', 'py38']
```
Comment (gaborbernat) [inline pyproject.toml:8]: Can reverse the order, I prefer to start with latest and go to smallest. Also, what do you think about making it merge friendly, and explode the list to be one version per line?

Comment (FFY00) [reply pyproject.toml:8]: I prefer it the other way :upside_down_face: but sure, doesn't bother me *much*.

File: tests/test_projectbuilder.py [hunk 3]
```diff
@@ -28,19 +30,13 @@
...
 
 
-DEFAULT_BACKEND = {
-    'build-backend': 'setuptools.build_meta:__legacy__',
-    'requires': [
-        'setuptools >= 40.8.0',
-        'wheel'
-    ]
-}
+DEFAULT_BACKEND = {'build-backend': 'setuptools.build_meta:__legacy__', 'requires': ['setuptools >= 40.8.0', 'wheel']}
```
Comment (gaborbernat) [inline tests/test_projectbuilder.py:39]: This is the one place where having it in separate lines is better, just add the trailing within the array and dictionary ``,`` and black will format it on separate lines.

File: tests/test_integration.py [hunk 4]
```diff
@@ -14,30 +14,14 @@
...
-    ('project'),
-    [
-        'python-build',
-        'pip',
-        'dateutil',
-        'Solaar',
-        os.path.join('flit', 'flit_core')
-    ]
-)
+@pytest.mark.parametrize(('project'), ['python-build', 'pip', 'dateutil', 'Solaar', os.path.join('flit', 'flit_core')])
```
Comment (gaborbernat) [inline tests/test_integration.py:17]: here to append the trailing comma to keep it on separate lines

File: tests/test_integration.py [hunk 5]
```diff
@@ -14,30 +14,14 @@
...
+    [[], ['-x', '--no-isolation']],
 )
 @pytest.mark.parametrize(
     ('call'),
-    [
-        None,  # via code
-        ['python', '-m', 'build'],  # module
-        ['python-build'],  # entrypoint
-    ]
+    [None, ['python', '-m', 'build'], ['python-build']],  # via code  # module  # entrypoint
```
Comment (gaborbernat) [inline tests/test_integration.py:24]: here to append the trailing comma to keep it on separate lines

**Review by gaborbernat — APPROVED** [review]
Otherwise looks 👍

File: tests/test_projectbuilder.py (hunk 3 above)
Comment (FFY00) [reply tests/test_projectbuilder.py:39]: I agree, and I dislike black doing this. Is there a workaround so that black doesn't do this?

Comment (FFY00) [reply tests/test_projectbuilder.py:39]: I had the `,` before but didn't work. Let me try again.

Comment (FFY00) [reply tests/test_projectbuilder.py:39]: Ah, it does work, sorry!

File: pyproject.toml (hunk 2 above)
Comment (gaborbernat) [reply pyproject.toml:8]: I prefer to focus on the new fancy versions, and prefer to forget the old ones at the end of the line 😆

File: tests/test_projectbuilder.py (hunk 3 above)
Comment (gaborbernat) [inline tests/test_projectbuilder.py:39]: Forgot this line?

File: tests/test_integration.py [hunk 6]
```diff
@@ -21,23 +21,20 @@
...
     ('call'),
     [
-        None,  # via code
-        ['python', '-m', 'build'],  # module
-        ['python-build'],  # entrypoint
-    ]
+        None,
+        ['python', '-m', 'build'],
+        ['python-build'],
+    ],  # via code  # module  # entrypoint
```
Comment (gaborbernat) [inline tests/test_integration.py:37]: You probably want to re-explode these comments 🤔

**Comment by FFY00** [comment]
Thanks for letting me know about the trailing `,` trick, it makes black much more usable for me! I went ahead and applied in some other places I though it made sense.

**Comment by gaborbernat** [comment]
> Thanks for letting me know about the trailing `,` trick,

Yeah, adding them also means that when you add a new entry you only need to change the newline, so one-line diff, instead of the usual 2 (add coma, plus new line content).

**Review by gaborbernat — APPROVED** [review]

**Comment by pganssle** [comment]
FWIW I support anything that makes it so that I can type my code in any syntactically valid way and have it automatically corrected to be in the project style. `black` or some variation thereof is perfectly acceptable to me.

I personally don't love the 127 character line length, because I generally code with 2 editors / terminal windows side-by-side, which is only ~100 characters with a reasonable font size, and it makes super wide lines annoyingly long. I'd rather either leave the default value or use 80.

That said, it doesn't really matter that much to me.

**Review by gaborbernat — APPROVED** [review]
